models/proposal/proposalForMessage.ts

```diff
@@ -0,0 +1,40 @@
+import { EmptyProposal, Proposal } from './proposal'
+import { ProposalReducer } from './proposalReducer'
+
+export function proposalForMessage(json: any): Proposal {
+  let proposal: Proposal = { ...EmptyProposal }
+  return updateProposalForMessage(proposal, json)
+}
+
+export function updateProposalForMessage(
+  proposal: Proposal,
+  json: any
+): Proposal {
+  let valid = true
```

Comment:
can valid be false? don't see a case now, just checking

Reply:
I was thinking we'd have some draft proposal updates where the message isn't valid but deleting for now